daemon/daemon.go

```diff
@@ -450,6 +450,20 @@ func NewDaemon(ctx context.Context, dp datapath.Datapath) (*Daemon, *endpointRes
 		return nil, nil, err
 	}
 
+	// Trigger refresh and update custom resource in the apiserver with all restored endpoints
```

Comment:
If I understand the intent of this code correctly, then you are adding this code because the following:
```
	store.refreshTrigger.TriggerWithReason("initial sync")
```

which also triggers a sync of the node is done too early and not blocking. Correct?

Instead of adding this code explicitly, please extend `pkg/trigger` with a function to wait for the first execution of the trigger and then invoke the existing trigger instead of introducing a parallel way of invoking the sync. That way, the metrics on trigger invocations remain correct.

Reply:
Thanks for the comments! I understand and absolutely agree with your idea that we should do all the sync actions from the existing trigger. I don't fully understand
```
please extend pkg/trigger with a function to wait for the first execution of the trigger and then invoke the existing trigger
```
Do you mean extending pkg/trigger with a function, say `TriggerAfter`, which waits for restoration of pods, `router` and `health` endpoints and then invoke the existing trigger?

And also replace the following:
```
	store.refreshTrigger.TriggerWithReason("initial sync")
```
with
```
	store.refreshTrigger.TriggerAfter(checkRestoration)
```
Is my understanding right?